lib/dndmonster.sty

```diff
@@ -124,62 +129,104 @@
+			\end{tabular}
+		}%
+		\\[0.4em] %adds space after table
+    \endgroup
 }
 
 \newkeycommand\details[skills=,
-                      damageimmunities=,
-                      savingthrows=,
-                      conditionimmunities=,
-                      damageresistances=,
-                      damagevulnerabilities=,
-                      senses=---,
-                      languages=---,
-                      challenge=0]{%
-	\ifcommandkey{savingthrows}{%
-		\begin{hangingpar}
+						damageimmunities=,
+						savingthrows=,
+						conditionimmunities=,
+						damageresistances=,
+						damagevulnerabilities=,
+						senses=---,
+						languages=---,
+						challenge=0]{%
+	\begingroup
+		\unskip\color{titlered}
+		\ifcommandkey{savingthrows}{%
+			\begin{hangingpar}
 			\textbf{\savesname} \commandkey{savingthrows}
-		\end{hangingpar}}{}
-	\ifcommandkey{skills}{%
-		\begin{hangingpar}
-			\textbf{\skillsname} \commandkey{skills}
-		\end{hangingpar}}{}
-	\ifcommandkey{damagevulnerabilities}{%
+			\end{hangingpar}}{}%
+		\ifcommandkey{skills}{%
+			\begin{hangingpar}
+				\textbf{\skillsname} \commandkey{skills}
+			\end{hangingpar}}{}%
+		\ifcommandkey{damagevulnerabilities}{%
+			\begin{hangingpar}
+				\textbf{\dvulname} \commandkey{damagevulnerabilities}
+			\end{hangingpar}}{}%
+		\ifcommandkey{damageresistances}{%
+			\begin{hangingpar}
+				\textbf{\dresname} \commandkey{damageresistances}
+			\end{hangingpar}}{}%
+		\ifcommandkey{damageimmunities}{%
+			\begin{hangingpar}
+				\textbf{\dimmname} \commandkey{damageimmunities}
+			\end{hangingpar}}{}%
+		\ifcommandkey{conditionimmunities}{%
+			\begin{hangingpar}
+				\textbf{\cimmname} \commandkey{conditionimmunities}
+			\end{hangingpar}}{}%
+		% These traits appear to always be present.
 		\begin{hangingpar}
-			\textbf{\dvulname} \commandkey{damagevulnerabilities}
-		\end{hangingpar}}{}
-	\ifcommandkey{damageresistances}{%
+			\textbf{\sensesname} \commandkey{senses}
+		\end{hangingpar}
 		\begin{hangingpar}
-			\textbf{\dresname} \commandkey{damageresistances}
-		\end{hangingpar}}{}
-	\ifcommandkey{damageimmunities}{%
+			\textbf{\languagesname} \commandkey{languages}
+		\end{hangingpar}
 		\begin{hangingpar}
-			\textbf{\dimmname} \commandkey{damageimmunities}
-		\end{hangingpar}}{}
-	\ifcommandkey{conditionimmunities}{%
-		\begin{hangingpar}
-			\textbf{\cimmname} \commandkey{conditionimmunities}
-		\end{hangingpar}}{}
-	% These traits appear to always be present.
-	\begin{hangingpar}
-		\textbf{\sensesname} \commandkey{senses}
-	\end{hangingpar}
-	\begin{hangingpar}
-		\textbf{\languagesname} \commandkey{languages}
-	\end{hangingpar}
-	\begin{hangingpar}
-		\textbf{\challengename} \commandkey{challenge}
-	\end{hangingpar}
-	\color{black}
+			\textbf{\challengename} \crexp{\commandkey{challenge}}
+		\end{hangingpar}%
+	\endgroup%
 }
+
+\newcommand\crexp[1]{%
+	\IfStrEqCase{#1}{%
+		{0}{#1 (0 XP)}%
+		{1/10}{0 (10 XP)}%
```

Comment:
This value is not very intuitive. What about an an interface like this?

```latex
\crexp[XP]{CR}
```

If the user wanted to print "0 (10 XP)", they could write `\crexp[10]{0}`. If they passed this as the `challenge` key, it would fall through and render as-is.

Something like this should work:

```latex
\NewDocumentCommand{\crexp}{o m}{%
  \IfValueTF{#1}{#2 (#1 XP)}{%
    \IfStrEqCase{#2}{%
      {0}{#2 (0 XP)}
      {1/8}{#2 (25 XP)}
      % ...
    }[#2]
  }
}
```

Reply:
I know the `0 (10 XP)` option is not intuitive, but it is an extreme corner case. Hardly anyone uses it. Making a `1/10` option seems like a better alternative to me.

I do not think the alternative you have proposed will work the way I want it to. The `challenge` keyword on the `\details` command now passes its value into the `\crexp` macro. This allows a user to pass just the CR to the `\details` command, and its current interface causes no breaking changes as anything other than a CR number is printed as is in the default case at the end of the switch statement. In its proposed implementation
```
\details[challenge=1/10]%
% and
\details[challenge=0 (10 XP)]%
```
both work, but changing `\crexp` to use an optional parameter would not allow it to be used behind the `challenge` keyword.

Challenge 0 monsters are the corner case as they have two possible XP values. There are 3 ways I can think of handling it.
1. Remove `0 (10 XP)` and require it to be written out.
2. Use an optional argument as you proposed above.
3. Have a non-standard CR that still makes a little sense as a special key.
The third option seems to be the only option that keeps the usage of the `\details` command unchanged even in the corner case.

Reply:
You're right, there's no nice way to pass in an optional parameter for the key. But what I meant was passing it through like this:

```latex
\details[challenge=\crexp[10]{0}]
```

The `\details` macro would receive `0 (10 XP)`. Just as though the user typed that themselves. It's sort of a combination of (1) and (2) in your message above.

Sidenote: should there be non-breaking spaces like this: `<cr>~(<xp>~XP)`?